### ISSUE TITLE
[JENKINS-46230] git-lfs not working in kubernetes container

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/kubernetes-plugin#1912]
<p>Trying to make use of the git-lfs addition to the checkout step, but it's failing.</p>

<p>The error I'm getting is:</p>

```
Fetching without tags
Fetching upstream changes from https://github.com/---
using GIT_ASKPASS to set credentials 
 > git fetch --tags --progress https://github.com/--- +refs/heads/master:refs/remotes/origin/master --depth=2
Checking out Revision 33d759e (master)
Commit message: "---"
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 33d759e

hudson.plugins.git.GitException: Command "git checkout -f 33d759e" returned status code 128:
stdout: 
stderr: git-lfs filter-process: line 1: git-lfs: not found
fatal: The remote end hung up unexpectedly

```

<p>The example pipeline code that is failing: </p>

<pre>
podTemplate(label: <span class="code-quote">"test"</span>, containers: [
    containerTemplate(name: <span class="code-quote">"golang"</span>, image: <span class="code-quote">"golang:1.8"</span>, ttyEnabled: <span class="code-keyword">true</span>, command: <span class="code-quote">"cat"</span>
    )
]) {
	node(<span class="code-quote">"test"</span>) {
		container(name: <span class="code-quote">"golang"</span>) {
			stage(<span class="code-quote">"checkout"</span>) {
				<span class="code-comment">// install git lfs on debian based pod
</span>				sh <span class="code-quote">"curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash"</span>
</span>				sh <span class="code-quote">"apt-get install git-lfs"</span>
				sh <span class="code-quote">"git lfs install"</span>
				checkout([
					$class: <span class="code-quote">"GitSCM"</span>,
					branches: scm.branches,
					extensions: scm.extensions + [
						[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>],
						[$class: <span class="code-quote">"GitLFSPull"</span>]
					],
					userRemoteConfigs: scm.userRemoteConfigs
				])
				sh <span class="code-quote">'ls -l'</span>
			}
		}
	}
}
 </pre>

<hr />
<p>If I log into the pod after git-lfs is installed and run `git clone` directly, it works and pulls the large file. My guess is that this problem is related to this bug/workaround, but I don't know how to implement this in the k8s/jenkins env<br/>
<a href="https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540</a></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek">jknurek</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230" target="_blank">git-lfs not working in kubernetes container</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: git-client-plugin, kubernetes-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2020-10-29T11:51:04+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Trying to make use of the git-lfs addition to the checkout step, but it's failing.</p>

<p>The error I'm getting is:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Fetching without tags
Fetching upstream changes from https://github.com/---
using GIT_ASKPASS to set credentials 
 &gt; git fetch --tags --progress https://github.com/--- +refs/heads/master:refs/remotes/origin/master --depth=2
Checking out Revision 33d759e (master)
Commit message: "---"
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 33d759e

hudson.plugins.git.GitException: Command "git checkout -f 33d759e" returned status code 128:
stdout: 
stderr: git-lfs filter-process: line 1: git-lfs: not found
fatal: The remote end hung up unexpectedly
</pre>
</div></div>

<p>The example pipeline code that is failing: </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">podTemplate(label: <span class="code-quote">"test"</span>, containers: [
    containerTemplate(name: <span class="code-quote">"golang"</span>, image: <span class="code-quote">"golang:1.8"</span>, ttyEnabled: <span class="code-keyword">true</span>, command: <span class="code-quote">"cat"</span>
    )
]) {
	node(<span class="code-quote">"test"</span>) {
		container(name: <span class="code-quote">"golang"</span>) {
			stage(<span class="code-quote">"checkout"</span>) {
				<span class="code-comment">// install git lfs on debian based pod
</span>				sh <span class="code-quote">"curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash"</span>
</span>				sh <span class="code-quote">"apt-get install git-lfs"</span>
				sh <span class="code-quote">"git lfs install"</span>
				checkout([
					$class: <span class="code-quote">"GitSCM"</span>,
					branches: scm.branches,
					extensions: scm.extensions + [
						[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>],
						[$class: <span class="code-quote">"GitLFSPull"</span>]
					],
					userRemoteConfigs: scm.userRemoteConfigs
				])
				sh <span class="code-quote">'ls -l'</span>
			}
		}
	}
}
 </pre>
</div></div>

<hr />
<p>If I log into the pod after git-lfs is installed and run `git clone` directly, it works and pulls the large file. My guess is that this problem is related to this bug/workaround, but I don't know how to implement this in the k8s/jenkins env
<a href="https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540</a></p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310633&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310633">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I don't understand <a href="https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540" class="external-link" target="_blank" rel="nofollow noopener">which work around</a> you believe would resolve your issue in your kubernetes container.  </p>

<p>That issue on GitHub seems to mention credential helpers, even though Linux machines (and kubernetes pods) rarely use credentials helpers.  Your pod does not seem to be defining a credentials helper, so I assume it is not a credential helper issue.</p>

<p>That issue also mentions case insensitive file name extensions, yet you said that the clone works when you perform a <tt>git clone</tt> from inside the kubernetes pod.  I would expect case insensitive file name extensions to cause that to break no matter what you did, since the typical Linux file system is case sensitive.</p>

<p>The message from the checkout step hints (to me) that the lfs command was not installed in a portion of the PATH defined for the checkout step.  That is quite odd, since I assume the preceding <tt>git lfs install</tt> is not reporting any errors.  Can you confirm that <tt>git lfs install</tt> reports that git lfs is installed?</p>

<p>Can you perform the <tt>git init; git remote add origin your_url; git fetch origin; git lfs fetch origin origin/master</tt> command from a shell step immediately prior to your checkout step?  </p>

<p>What is the output of <tt>git --version</tt> on your kubernetes instances?  Is the git new enough to support LFS?</p>
</pre>
</details>

</ul>
</details>

<p>I don't understand <a href="https://github.com/git-lfs/git-lfs/issues/1821#issuecomment-272889540" class="external-link" target="_blank" rel="nofollow noopener">which work around</a> you believe would resolve your issue in your kubernetes container.  </p>

<p>That issue on GitHub seems to mention credential helpers, even though Linux machines (and kubernetes pods) rarely use credentials helpers.  Your pod does not seem to be defining a credentials helper, so I assume it is not a credential helper issue.</p>

<p>That issue also mentions case insensitive file name extensions, yet you said that the clone works when you perform a <tt>git clone</tt> from inside the kubernetes pod.  I would expect case insensitive file name extensions to cause that to break no matter what you did, since the typical Linux file system is case sensitive.</p>

<p>The message from the checkout step hints (to me) that the lfs command was not installed in a portion of the PATH defined for the checkout step.  That is quite odd, since I assume the preceding <tt>git lfs install</tt> is not reporting any errors.  Can you confirm that <tt>git lfs install</tt> reports that git lfs is installed?</p>

<p>Can you perform the <tt>git init; git remote add origin your_url; git fetch origin; git lfs fetch origin origin/master</tt> command from a shell step immediately prior to your checkout step?  </p>

<p>What is the output of <tt>git --version</tt> on your kubernetes instances?  Is the git new enough to support LFS?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310826&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310826">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I tried running <tt>git init; git remote add origin your_url; git fetch origin; git lfs fetch origin origin/master</tt> but the </p>

<p><tt>git fetch origin</tt> was not responding. Because I have other problems in later steps trying to use git's ssh connection, and have a work-around for it, I wrapped the calls and below are the results for the following commands:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">	sh <span class="code-quote">"curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash"</span>
</span>	sh <span class="code-quote">"apt-get install git-lfs"</span>
	sh <span class="code-quote">"git lfs install"</span>
	sh <span class="code-quote">"git init"</span>
	sh <span class="code-quote">"git remote add origin git@github.com:repo.git"</span>
	sh <span class="code-quote">"chmod 400 /root/.ssh/ssh-privatekey"</span>
	sh <span class="code-quote">"echo \"</span>Host github.com\n\tStrictHostKeyChecking no\n\<span class="code-quote">" &gt; /root/.ssh/config"</span>
	sh <span class="code-quote">"ssh-agent bash -c <span class="code-quote">'ssh-add /root/.ssh/ssh-privatekey; git fetch -v --depth=2 origin'</span>"</span>
	sh <span class="code-quote">"ssh-agent bash -c <span class="code-quote">'ssh-add /root/.ssh/ssh-privatekey; git lfs fetch origin origin/master'</span>"</span>
 </pre>
</div></div>
<hr />
<p>oh, and the git version in this container is: 2.1.4 </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">+ bash
+ curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh
</span>Detected operating system as debian/8.
Checking <span class="code-keyword">for</span> curl...
Detected curl...
Running apt-get update... done.
Installing debian-archive-keyring which is needed <span class="code-keyword">for</span> installing 
apt-transport-https on many Debian systems.
Installing apt-transport-https... done.
Installing /etc/apt/sources.list.d/github_git-lfs.list...done.
Importing packagecloud gpg key... done.
Running apt-get update... done.

The repository is setup! You can now install packages.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ apt-get install git-lfs
Reading <span class="code-keyword">package</span> lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  git-lfs
0 upgraded, 1 newly installed, 0 to remove and 36 not upgraded.
Need to get 2766 kB of archives.
After <span class="code-keyword">this</span> operation, 11.3 MB of additional disk space will be used.
Get:1 https:<span class="code-comment">//packagecloud.io/github/git-lfs/debian/ jessie/main git-lfs amd64 2.2.1 [2766 kB]
</span>debconf: delaying <span class="code-keyword">package</span> configuration, since apt-utils is not installed
Fetched 2766 kB in 1s (2049 kB/s)
Selecting previously unselected <span class="code-keyword">package</span> git-lfs.
(Reading database ... 
(Reading database ... 100%
(Reading database ... 14788 files and directories currently installed.)
Preparing to unpack .../git-lfs_2.2.1_amd64.deb ...
Unpacking git-lfs (2.2.1) ...
Setting up git-lfs (2.2.1) ...
Git LFS initialized.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git lfs install
Git LFS initialized.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git init
Initialized empty Git repository in /home/jenkins/workspace/repo_master-FZOPC6NFFVCNA/.git/
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git remote add origin git@github.com:repo.git
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ chmod 400 /root/.ssh/ssh-privatekey
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ echo Host github.com
	StrictHostKeyChecking no

[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ssh-agent bash -c ssh-add /root/.ssh/ssh-privatekey; git fetch -v --depth=2 origin
Identity added: /root/.ssh/ssh-privatekey (rsa w/o comment)
Warning: Permanently added <span class="code-quote">'github.com,192.30.253.112'</span> (RSA) to the list of known hosts.
From github.com:repo
 * [<span class="code-keyword">new</span> branch]      feature/16415-docker-fix -&gt; origin/feature/16415-docker-fix
 * [<span class="code-keyword">new</span> branch]      master     -&gt; origin/master
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ssh-agent bash -c ssh-add /root/.ssh/ssh-privatekey; git lfs fetch origin origin/master
Identity added: /root/.ssh/ssh-privatekey (rsa w/o comment)
Fetching origin/master

Git LFS: (0 of 1 files) 0 B / 116.62 MB                                        
Git LFS: (1 of 1 files) 116.62 MB / 116.62 MB                                  
[Pipeline] checkout
 &gt; git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 &gt; git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>ERROR: Error fetching remote repo <span class="code-quote">'origin'</span>
hudson.plugins.git.GitException: Failed to fetch from https:<span class="code-comment">//github.com/repo.git
</span>	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: hudson.plugins.git.GitException: Command <span class="code-quote">"git config remote.origin.url https:<span class="code-comment">//github.com/repo.git"</span> returned status code 255:
</span>stdout: 
stderr: error: could not lock config file .git/config: Permission denied

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1924)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1892)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1888)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1533)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1545)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.setRemoteUrl(CliGitAPIImpl.java:1179)
	at hudson.plugins.git.GitAPI.setRemoteUrl(GitAPI.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:895)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:870)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:829)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:336)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:94)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
	at ......remote call to JNLP4-connect connection from 10.132.5.130/10.132.5.130:42512(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1545)
	at hudson.remoting.UserResponse.retrieve(UserRequest.java:253)
	at hudson.remoting.Channel.call(Channel.java:830)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:257)
	at com.sun.proxy.$Proxy104.setRemoteUrl(Unknown Source)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl.setRemoteUrl(RemoteGitImpl.java:295)
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:805)
	... 13 more

</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I tried running <tt>git init; git remote add origin your_url; git fetch origin; git lfs fetch origin origin/master</tt> but the </p>

<p><tt>git fetch origin</tt> was not responding. Because I have other problems in later steps trying to use git's ssh connection, and have a work-around for it, I wrapped the calls and below are the results for the following commands:</p>

<pre>
	sh <span class="code-quote">"curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash"</span>
</span>	sh <span class="code-quote">"apt-get install git-lfs"</span>
	sh <span class="code-quote">"git lfs install"</span>
	sh <span class="code-quote">"git init"</span>
	sh <span class="code-quote">"git remote add origin git@​github.com:repo.git"</span>
	sh <span class="code-quote">"chmod 400 /root/.ssh/ssh-privatekey"</span>
	sh <span class="code-quote">"echo \"</span>Host github.com\n\tStrictHostKeyChecking no\n\<span class="code-quote">" > /root/.ssh/config"</span>
	sh <span class="code-quote">"ssh-agent bash -c <span class="code-quote">'ssh-add /root/.ssh/ssh-privatekey; git fetch -v --depth=2 origin'</span>"</span>
	sh <span class="code-quote">"ssh-agent bash -c <span class="code-quote">'ssh-add /root/.ssh/ssh-privatekey; git lfs fetch origin origin/master'</span>"</span>
 </pre>
<hr />
<p>oh, and the git version in this container is: 2.1.4 </p>

<pre>
+ bash
+ curl -s https:<span class="code-comment">//packagecloud.io/install/repositories/github/git-lfs/script.deb.sh
</span>Detected operating system as debian/8.
Checking <span class="code-keyword">for</span> curl...
Detected curl...
Running apt-get update... done.
Installing debian-archive-keyring which is needed <span class="code-keyword">for</span> installing 
apt-transport-https on many Debian systems.
Installing apt-transport-https... done.
Installing /etc/apt/sources.list.d/github_git-lfs.list...done.
Importing packagecloud gpg key... done.
Running apt-get update... done.

The repository is setup! You can now install packages.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ apt-get install git-lfs
Reading <span class="code-keyword">package</span> lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  git-lfs
0 upgraded, 1 newly installed, 0 to remove and 36 not upgraded.
Need to get 2766 kB of archives.
After <span class="code-keyword">this</span> operation, 11.3 MB of additional disk space will be used.
Get:1 https:<span class="code-comment">//packagecloud.io/github/git-lfs/debian/ jessie/main git-lfs amd64 2.2.1 [2766 kB]
</span>debconf: delaying <span class="code-keyword">package</span> configuration, since apt-utils is not installed
Fetched 2766 kB in 1s (2049 kB/s)
Selecting previously unselected <span class="code-keyword">package</span> git-lfs.
(Reading database ... 
(Reading database ... 100%
(Reading database ... 14788 files and directories currently installed.)
Preparing to unpack .../git-lfs_2.2.1_amd64.deb ...
Unpacking git-lfs (2.2.1) ...
Setting up git-lfs (2.2.1) ...
Git LFS initialized.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git lfs install
Git LFS initialized.
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git init
Initialized empty Git repository in /home/jenkins/workspace/repo_master-FZOPC6NFFVCNA/.git/
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ git remote add origin git@​github.com:repo.git
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ chmod 400 /root/.ssh/ssh-privatekey
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ echo Host github.com
	StrictHostKeyChecking no

[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ssh-agent bash -c ssh-add /root/.ssh/ssh-privatekey; git fetch -v --depth=2 origin
Identity added: /root/.ssh/ssh-privatekey (rsa w/o comment)
Warning: Permanently added <span class="code-quote">'github.com,192.30.253.112'</span> (RSA) to the list of known hosts.
From github.com:repo
 * [<span class="code-keyword">new</span> branch]      feature/16415-docker-fix -> origin/feature/16415-docker-fix
 * [<span class="code-keyword">new</span> branch]      master     -> origin/master
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ssh-agent bash -c ssh-add /root/.ssh/ssh-privatekey; git lfs fetch origin origin/master
Identity added: /root/.ssh/ssh-privatekey (rsa w/o comment)
Fetching origin/master

Git LFS: (0 of 1 files) 0 B / 116.62 MB
Git LFS: (1 of 1 files) 116.62 MB / 116.62 MB  
[Pipeline] checkout
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>ERROR: Error fetching remote repo <span class="code-quote">'origin'</span>
hudson.plugins.git.GitException: Failed to fetch from https:<span class="code-comment">//github.com/repo.git
</span>	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: hudson.plugins.git.GitException: Command <span class="code-quote">"git config remote.origin.url https:<span class="code-comment">//github.com/repo.git"</span> returned status code 255:
</span>stdout: 
stderr: error: could not lock config file .git/config: Permission denied

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1924)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1892)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1888)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1533)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1545)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.setRemoteUrl(CliGitAPIImpl.java:1179)
	at hudson.plugins.git.GitAPI.setRemoteUrl(GitAPI.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:895)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:870)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:829)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:336)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:94)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
	at ......remote call to JNLP4-connect connection from 10.132.5.130/10.132.5.130:42512(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1545)
	at hudson.remoting.UserResponse.retrieve(UserRequest.java:253)
	at hudson.remoting.Channel.call(Channel.java:830)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:257)
	at com.sun.proxy.$Proxy104.setRemoteUrl(Unknown Source)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl.setRemoteUrl(RemoteGitImpl.java:295)
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:805)
	... 13 more

</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310827&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310827">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The message </p>

<blockquote><p>could not lock config file .git/config: Permission denied</p></blockquote>

<p>would usually indicate that the wrong user owns that file, or the wrong ownership or permissions have been assigned to that file or the directory which contains the file.</p>
</pre>
</details>

</ul>
</details>

<p>The message </p>

<blockquote><p>could not lock config file .git/config: Permission denied</p></blockquote>

<p>would usually indicate that the wrong user owns that file, or the wrong ownership or permissions have been assigned to that file or the directory which contains the file.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310946&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310946">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">+ ls -l .git/config
-rwxr--r-- 1 root root 210 Aug 18 08:59 .git/config</pre>
</div></div>
<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">+ whoami
root</pre>
</div></div>
</pre>
</details>

</ul>
</details>


<pre>
+ ls -l .git/config
-rwxr--r-- 1 root root 210 Aug 18 08:59 .git/config</pre>
<p> </p>

<pre>
+ whoami
root</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310947&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310947">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>more specifically, this is after installing git lfs, and running init/fetch:</p>

<p>the pipeline steps:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">	sh <span class="code-quote">"ls -l .git/"</span>
	sh <span class="code-quote">"ls -al"</span>
	sh <span class="code-quote">"whoami"</span>
	checkout([
		$class: <span class="code-quote">"GitSCM"</span>,
		branches: scm.branches,
		extensions: scm.extensions + [
			[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>],
			[$class: <span class="code-quote">"GitLFSPull"</span>]
		],
		userRemoteConfigs: scm.userRemoteConfigs
	])
 </pre>
</div></div>
<p>and the output:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">+ ls -l .git/
total 48
-rw-r--r-- 1 root root  938 Aug 18 09:08 FETCH_HEAD
-rw-r--r-- 1 root root   23 Aug 18 09:08 HEAD
drwxr-xr-x 2 root root 4096 Aug 18 09:08 branches
-rwxr--r-- 1 root root  210 Aug 18 09:08 config
-rw-r--r-- 1 root root   73 Aug 18 09:08 description
drwxr-xr-x 2 root root 4096 Aug 18 09:08 hooks
drwxr-xr-x 2 root root 4096 Aug 18 09:08 info
drwxr-xr-x 4 root root 4096 Aug 18 09:08 lfs
drwxr-xr-x 3 root root 4096 Aug 18 09:08 logs
drwxr-xr-x 4 root root 4096 Aug 18 09:08 objects
drwxr-xr-x 5 root root 4096 Aug 18 09:08 refs
-rw-r--r-- 1 root root  287 Aug 18 09:08 shallow
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 12
drwxr-xr-x 3 10000 10000 4096 Aug 18 09:08 .
drwxr-xr-x 4 10000 10000 4096 Aug 18 09:08 ..
drwxr-xr-x 9 root  root  4096 Aug 18 09:08 .git
-rw------- 1 root  root     0 Aug 18 09:08 nohup.out
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ whoami
root
[Pipeline] checkout
 &gt; git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 &gt; git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>ERROR: Error fetching remote repo <span class="code-quote">'origin'</span>
hudson.plugins.git.GitException: Failed to fetch from https:<span class="code-comment">//github.com/repo.git
</span>	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: hudson.plugins.git.GitException: Command <span class="code-quote">"git config remote.origin.url https:<span class="code-comment">//github.com/repo.git"</span> returned status code 255:
</span>stdout: 
stderr: error: could not lock config file .git/config: Permission denied
 </pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>more specifically, this is after installing git lfs, and running init/fetch:</p>

<p>the pipeline steps:</p>

<pre>
	sh <span class="code-quote">"ls -l .git/"</span>
	sh <span class="code-quote">"ls -al"</span>
	sh <span class="code-quote">"whoami"</span>
	checkout([
		$class: <span class="code-quote">"GitSCM"</span>,
		branches: scm.branches,
		extensions: scm.extensions + [
			[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>],
			[$class: <span class="code-quote">"GitLFSPull"</span>]
		],
		userRemoteConfigs: scm.userRemoteConfigs
	])
 </pre>
<p>and the output:</p>

<pre>
+ ls -l .git/
total 48
-rw-r--r-- 1 root root  938 Aug 18 09:08 FETCH_HEAD
-rw-r--r-- 1 root root   23 Aug 18 09:08 HEAD
drwxr-xr-x 2 root root 4096 Aug 18 09:08 branches
-rwxr--r-- 1 root root  210 Aug 18 09:08 config
-rw-r--r-- 1 root root   73 Aug 18 09:08 description
drwxr-xr-x 2 root root 4096 Aug 18 09:08 hooks
drwxr-xr-x 2 root root 4096 Aug 18 09:08 info
drwxr-xr-x 4 root root 4096 Aug 18 09:08 lfs
drwxr-xr-x 3 root root 4096 Aug 18 09:08 logs
drwxr-xr-x 4 root root 4096 Aug 18 09:08 objects
drwxr-xr-x 5 root root 4096 Aug 18 09:08 refs
-rw-r--r-- 1 root root  287 Aug 18 09:08 shallow
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 12
drwxr-xr-x 3 10000 10000 4096 Aug 18 09:08 .
drwxr-xr-x 4 10000 10000 4096 Aug 18 09:08 ..
drwxr-xr-x 9 root  root  4096 Aug 18 09:08 .git
-rw------- 1 root  root     0 Aug 18 09:08 nohup.out
[Pipeline] sh
[repo_master-FZOPC6NFFVCNA] Running shell script
+ whoami
root
[Pipeline] checkout
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>ERROR: Error fetching remote repo <span class="code-quote">'origin'</span>
hudson.plugins.git.GitException: Failed to fetch from https:<span class="code-comment">//github.com/repo.git
</span>	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: hudson.plugins.git.GitException: Command <span class="code-quote">"git config remote.origin.url https:<span class="code-comment">//github.com/repo.git"</span> returned status code 255:
</span>stdout: 
stderr: error: could not lock config file .git/config: Permission denied
 </pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=310978&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310978">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry, but I don't know how to diagnose the permissions issue which is causing git to be unable to lock the .git/config file.  It could be a problem from the underlying file system where the volume is mounted.  It could be a permissions issue on the file.  It could be an ownership issue on the file.</p>

<p>You might create a temporary subdirectory in that repository, change to that subdirectory, perform a git init, and a git config command in that subdirectory (to create the .git/config file), then compare the ownership and permissions on the resulting file to the problem file and its parent directory.</p>
</pre>
</details>

</ul>
</details>

<p>Sorry, but I don't know how to diagnose the permissions issue which is causing git to be unable to lock the .git/config file.  It could be a problem from the underlying file system where the volume is mounted.  It could be a permissions issue on the file.  It could be an ownership issue on the file.</p>

<p>You might create a temporary subdirectory in that repository, change to that subdirectory, perform a git init, and a git config command in that subdirectory (to create the .git/config file), then compare the ownership and permissions on the resulting file to the problem file and its parent directory.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=311081&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-311081">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I don't think this has anything to do with the container but with either the kubernetes plugin OR the checkout step (or both). Because when I run this simplified step you can see that Jenkins is using a different user to checkout the code:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">container(name: <span class="code-quote">"golang"</span>) {
	stage(<span class="code-quote">"checkout"</span>) {
		sh <span class="code-quote">"ls -al"</span>
		sh <span class="code-quote">"whoami"</span>
		checkout([
			$class: <span class="code-quote">"GitSCM"</span>,
			branches: scm.branches,
			extensions: scm.extensions + [
				[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>]
			],
			userRemoteConfigs: scm.userRemoteConfigs
		])
		sh <span class="code-quote">"ls -al"</span>
	}
}
 </pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 8
drwxr-xr-x 2 10000 10000 4096 Aug 21 07:28 .
drwxr-xr-x 4 10000 10000 4096 Aug 21 07:28 ..
-rw------- 1 root  root     0 Aug 21 07:28 nohup.out
[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ whoami
root
[Pipeline] checkout
Cloning the remote Git repository
Cloning with configured refspecs honoured and without tags
Using shallow clone
shallow clone depth 2
Cloning repository https:<span class="code-comment">//github.com/repo.git
</span> &gt; git init /home/jenkins/workspace/rep_master-FZOPC6NFFVCNA # timeout=10
Fetching upstream changes from https:<span class="code-comment">//github.com/repo.git
</span> &gt; git --version # timeout=10
using GIT_ASKPASS to set credentials 
 &gt; git fetch --no-tags --progress https:<span class="code-comment">//github.com/repo.git +refs/heads/master:refs/remotes/origin/master --depth=2
</span> &gt; git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span> &gt; git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
 &gt; git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>Fetching without tags
Fetching upstream changes from https:<span class="code-comment">//github.com/repo.git
</span>using GIT_ASKPASS to set credentials 
 &gt; git fetch --tags --progress https:<span class="code-comment">//github.com/repo.git +refs/heads/master:refs/remotes/origin/master --depth=2
</span>Checking out Revision 33d759ea0 (master)
Commit message: <span class="code-quote">"...."</span>
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 33d759ea0
 &gt; git rev-list 33d759ea0 # timeout=10
[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 112
drwxr-xr-x 7 10000 10000 4096 Aug 21 07:28 .
drwxr-xr-x 4 10000 10000 4096 Aug 21 07:28 ..
-rw-r--r-- 1 10000 10000  131 Aug 21 07:28 .dockerignore
drwxr-xr-x 8 10000 10000 4096 Aug 21 07:28 .git
-rw-r--r-- 1 10000 10000   71 Aug 21 07:28 .gitattributes
-rw-r--r-- 1 10000 10000  120 Aug 21 07:28 .gitignore
-rw-r--r-- 1 10000 10000 1189 Aug 21 07:28 .gitlab-ci.yml
-rw-r--r-- 1 10000 10000  522 Aug 21 07:28 Dockerfile
-rw-r--r-- 1 10000 10000  197 Aug 21 07:28 Jenkinsfile
-rw-r--r-- 1 10000 10000 6445 Aug 21 07:28 Makefile
-rw-r--r-- 1 10000 10000 4186 Aug 21 07:28 README.md
-rw-r--r-- 1 10000 10000 2772 Aug 21 07:28 docker-compose.yml
-rwxr-xr-x 1 10000 10000 1948 Aug 21 07:28 docker-entrypoint.sh
-rw-r--r-- 1 10000 10000 8577 Aug 21 07:28 glide.lock
-rw-r--r-- 1 10000 10000 3481 Aug 21 07:28 glide.yaml
-rw------- 1 root  root     0 Aug 21 07:28 nohup.out
drwxr-xr-x 2 10000 10000 4096 Aug 21 07:28 scripts
[Pipeline] }
</pre>
</div></div>
<p>so in our previous example, when we use the `sh` step to run git init/fetch, the config is created as the root user, but then the checkout step is using the `10000` user </p>
</pre>
</details>

</ul>
</details>

<p>I don't think this has anything to do with the container but with either the kubernetes plugin OR the checkout step (or both). Because when I run this simplified step you can see that Jenkins is using a different user to checkout the code:</p>

<pre>
container(name: <span class="code-quote">"golang"</span>) {
	stage(<span class="code-quote">"checkout"</span>) {
		sh <span class="code-quote">"ls -al"</span>
		sh <span class="code-quote">"whoami"</span>
		checkout([
			$class: <span class="code-quote">"GitSCM"</span>,
			branches: scm.branches,
			extensions: scm.extensions + [
				[$class: <span class="code-quote">"CloneOption"</span>, depth: 2, shallow: <span class="code-keyword">true</span>]
			],
			userRemoteConfigs: scm.userRemoteConfigs
		])
		sh <span class="code-quote">"ls -al"</span>
	}
}
 </pre>

<pre>
[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 8
drwxr-xr-x 2 10000 10000 4096 Aug 21 07:28 .
drwxr-xr-x 4 10000 10000 4096 Aug 21 07:28 ..
-rw------- 1 root  root     0 Aug 21 07:28 nohup.out
[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ whoami
root
[Pipeline] checkout
Cloning the remote Git repository
Cloning with configured refspecs honoured and without tags
Using shallow clone
shallow clone depth 2
Cloning repository https:<span class="code-comment">//github.com/repo.git
</span> > git init /home/jenkins/workspace/rep_master-FZOPC6NFFVCNA # timeout=10
Fetching upstream changes from https:<span class="code-comment">//github.com/repo.git
</span> > git --version # timeout=10
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --progress https:<span class="code-comment">//github.com/repo.git +refs/heads/master:refs/remotes/origin/master --depth=2
</span> > git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span> > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
 > git config remote.origin.url https:<span class="code-comment">//github.com/repo.git # timeout=10
</span>Fetching without tags
Fetching upstream changes from https:<span class="code-comment">//github.com/repo.git
</span>using GIT_ASKPASS to set credentials 
 > git fetch --tags --progress https:<span class="code-comment">//github.com/repo.git +refs/heads/master:refs/remotes/origin/master --depth=2
</span>Checking out Revision 33d759ea0 (master)
Commit message: <span class="code-quote">"...."</span>
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 33d759ea0
 > git rev-list 33d759ea0 # timeout=10
[Pipeline] sh
[rep_master-FZOPC6NFFVCNA] Running shell script
+ ls -al
total 112
drwxr-xr-x 7 10000 10000 4096 Aug 21 07:28 .
drwxr-xr-x 4 10000 10000 4096 Aug 21 07:28 ..
-rw-r--r-- 1 10000 10000  131 Aug 21 07:28 .dockerignore
drwxr-xr-x 8 10000 10000 4096 Aug 21 07:28 .git
-rw-r--r-- 1 10000 10000   71 Aug 21 07:28 .gitattributes
-rw-r--r-- 1 10000 10000  120 Aug 21 07:28 .gitignore
-rw-r--r-- 1 10000 10000 1189 Aug 21 07:28 .gitlab-ci.yml
-rw-r--r-- 1 10000 10000  522 Aug 21 07:28 Dockerfile
-rw-r--r-- 1 10000 10000  197 Aug 21 07:28 Jenkinsfile
-rw-r--r-- 1 10000 10000 6445 Aug 21 07:28 Makefile
-rw-r--r-- 1 10000 10000 4186 Aug 21 07:28 README.md
-rw-r--r-- 1 10000 10000 2772 Aug 21 07:28 docker-compose.yml
-rwxr-xr-x 1 10000 10000 1948 Aug 21 07:28 docker-entrypoint.sh
-rw-r--r-- 1 10000 10000 8577 Aug 21 07:28 glide.lock
-rw-r--r-- 1 10000 10000 3481 Aug 21 07:28 glide.yaml
-rw------- 1 root  root     0 Aug 21 07:28 nohup.out
drwxr-xr-x 2 10000 10000 4096 Aug 21 07:28 scripts
[Pipeline] }
</pre>
<p>so in our previous example, when we use the `sh` step to run git init/fetch, the config is created as the root user, but then the checkout step is using the `10000` user </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> csanchez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=359812&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-359812">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>my guess is that shell commands are executed inside the golang container as expected (inside container("golang")) but the checkout step is running in the default jnlp container</p>
</pre>
</details>

</ul>
</details>

<p>my guess is that shell commands are executed inside the golang container as expected (inside container("golang")) but the checkout step is running in the default jnlp container</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=359923&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-359923">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>yes, I think you're right.</p>

<p> </p>

<p>I was able to recently workaround this by using a custom jnlp image that has git lfs installed. Which to me implies that this issue is merely just blocked by: <a href="https://github.com/jenkinsci/docker-slave/pull/39" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker-slave/pull/39</a></p>
</pre>
</details>

</ul>
</details>

<p>yes, I think you're right.</p>

<p> </p>

<p>I was able to recently workaround this by using a custom jnlp image that has git lfs installed. Which to me implies that this issue is merely just blocked by: <a href="https://github.com/jenkinsci/docker-slave/pull/39" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker-slave/pull/39</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46230?focusedId=400175&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-400175">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Closing as a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-30600" title="git isn't run inside build container" class="issue-link" data-issue-key="JENKINS-30600">JENKINS-30600</a>. </p>
</pre>
</details>

</ul>
</details>

<p>Closing as a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-30600" title="git isn't run inside build container" class="issue-link" data-issue-key="JENKINS-30600">JENKINS-30600</a>. </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/kubernetes-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-47389%22">JENKINS-47389</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/kubernetes-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-30600%22">JENKINS-30600</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/kubernetes-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-46055%22">JENKINS-46055</a>]</i>
